src/branching.rs

```diff
@@ -0,0 +1,62 @@
+
+#[derive(Clone)]
+pub enum BranchSpec {
+    ByBranchId(u32),
+    ByBranchName(String),
+}
+impl <'a> BranchSpec {
+    // for ingesting strings in the shape of "branch=name" or "branchId=2"
+    pub fn parse(branch_spec_input: String) -> Result<BranchSpec, &'a str> {
+        let branch_spec_string = branch_spec_input.to_string();
+        if branch_spec_string.starts_with("branchId=") {
+            let split_branch_id= branch_spec_string.split_once('=');
+            match split_branch_id {
+                Some(("branchId", id)) => {
+                    let branch_id: u32 = id
+                        .parse::<u32>()
+                        .map_err(|_| "project-id should be a number")?;
+                        Ok(BranchSpec::ByBranchId(branch_id))
```

Comment:
You could maybe inline this by using `.and_then` instead of the `?`, e.g. `.and_then(|branch_id| BranchSpec::ByBranchId(branch_id))`

If not, at least fix the indentation (the Ok is +1 tab from where it should be)

Reply:
Follow-up: I should've suggested `map` instead of `and_then`